sympy/solvers/tests/test_ode.py

```diff
@@ -3004,10 +3004,20 @@ def test_2nd_power_series_ordinary():
         + 1) + O(x**6))
 
     eq = f(x).diff(x, 2) + x*f(x)
-    assert classify_ode(eq) == ('2nd_power_series_ordinary',)
-    assert dsolve(eq, n=7) == Eq(f(x), C2*(
+    assert classify_ode(eq) == ('2nd_linear_airy', '2nd_power_series_ordinary')
+    assert dsolve(eq, n=7, hint='2nd_power_series_ordinary') == Eq(f(x), C2*(
         x**6/180 - x**3/6 + 1) + C1*x*(-x**3/12 + 1) + O(x**7))
 
+def test_Airy_equation():
+    C1, C2 = symbols("C1 C2")
+    eq = f(x).diff(x, 2) - x*f(x)
+    assert classify_ode(eq) == ("2nd_linear_airy",'2nd_power_series_ordinary')
+    assert dsolve(eq) == Eq(f(x), C1*airyai(x) + C2*airybi(x))
+    eq = f(x).diff(x, 2) + 2*x*f(x)
+    assert classify_ode(eq) == ("2nd_linear_airy",'2nd_power_series_ordinary')
+    assert dsolve(eq) == Eq(f(x), C1*airyai(-2**(S(1)/3)*x) + C2*airybi(-2**(S(1)/3)*x))
+
+
 
 def test_2nd_power_series_regular():
     # FIXME: Maybe there should be a way to check series solutions
```

Comment:
checkodesol doesn't work with series solution

Reply:
That's correct. If you look throughout the file though almost every call to dsolve is matched with checkodesol. A few aren't and that is either because the value returned by dsolve is incorrect or because of a problem with checkodesol.

 You should do the same with the tests you are adding or changing.

Reply:
Okk